### DISCUSSION TITLE
Printable report for hits/downloads

### DISCUSSION THREAD

**Discussion by bcordis** [Joomla-Bible-Study/Proclaim#848] [Ideas]
James House:
How about a printable report to show hits/downloads for media files and hits for series and/or studies?
I would like to have a nice report to give to our pastor to show the possible impact the online series/sermons are having. 

http://www.joomlabiblestudy.org/forum/feature-requests-jbs-8/3552-printable-report-for-hits-downloads.html#3744

**Comment by genu** [comment]
We don't collect a lot of metrics. The only thing we collect are **plays** and **downloads** for media files. For more complete analytics, users should just use specialized solutions like Google analytics.

We could, possibly, implement a minimal report just based on those 2 metrics (plays and downloads), but we're not capturing when users play or download media.

The only information we can conclude from the information that we currently gather, for example are:
- File X was download/played y times"
- Show the most popular studies
- The most popular media files for a particular study
- Most popular series
- Most popular speakers

More useful analytics would be to include time frames for example:
- Number of downloads/plays in a particular month

How complex reports do we want to generate?

**Comment by tomfuller2** [comment]
The other problem is that we only capture metrics when someone has used the component. This doesn't count anything accessed directly from a podcast, for instance.

**Comment by bcordis** [comment]
We could look into allowing sub Analytics Number for just bible study system like form or there's. or we creat a doc showing how others can use google Analytics to get report on bible study system.

**Comment by genu** [comment]
If users use google analytics, I think it takes care of everything. It will measure hits to a particular study page. The only thing that it will not do is measure plays via the podcast (Doesn't itunes have some sort of analytics feature for this though?).

I think that currently, the _plays_ and _downloads_ that we are capturing are not meant to be used for analytical purposes. Wasn't the original purpose to just make it possible to display the view/play count for studies on the frontend?

Correct me if I'm wrong.

**Comment by tomfuller2** [comment]
Yes, you are right. I don't think we should get too crazy on this report. But the info could be sliced and diced coupling plays/downloads with dates, series, teachers, etc.

**Comment by genu** [comment]
like @tomfuller2  said, we can just combine those matrics. One thing we can do is just elaborate on the metrics that are already shown on the main component page (the admin landing page), and just add a button to print. Of course, we'll generate a formal printable version of the landing page with the metrics. That way, we don't have to complicate ourselves with introducing any new logic or models/controls/views, etc just to have a simple report.

**Comment by bcordis** [comment]
Implemented in PR #1114.

The analytics dashboard now has a **Print Report** button next to the CSV export. It opens a clean, paper-optimized popup with:

- Site name and logo in the header (uses the default study image from Proclaim settings)
- KPI summary boxes (all-time and period totals)
- Clean, zebra-striped tables for all data sections
- Support for all drill-down levels (overview, series list, series detail, message detail, media type)
- Campus name when filtered
- Landscape `@page` orientation for wide tables
- Works with browser Print/Save as PDF

No Chart.js canvases or interactive UI — just numbers a pastor can read.
